atc/db/resource.go

```diff
@@ -296,6 +297,35 @@ func (r *resource) SaveUncheckedVersion(version atc.Version, metadata ResourceCo
 	return newVersion, tx.Commit()
 }
 
+func (r *resource) UpdateMetadata(version atc.Version, metadata ResourceConfigMetadataFields) (bool, error) {
+	versionJSON, err := json.Marshal(version)
+	if err != nil {
+		return false, err
+	}
+
+	metadataJSON, err := json.Marshal(metadata)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = psql.Update("resource_config_versions").
```

Comment:
:thought_balloon: this seems a little iffy/imprecise but it's probably fine.

just thinking about the case where the pipeline is re-configured while the build is in-flight, with a different resource config, which happens to produce the same version identifiers, but runs with a different type or something such that the metadata is not accurate to the new resource type

the likelihood and impact of this kind of edge case is so low that it doesn't seem like something to worry about

Reply:
would matching on `version_md5` be faster here?

Reply:
Yeah good call on the `version_md5`. Updated.